client/src/components/CourseView.tsx

```diff
@@ -36,10 +34,8 @@ function CourseView(): JSX.Element {
     useState(null);
   const [assessmentModels, setAssessmentModels]: State<Array<AssessmentModelData>> =
     useState([]);
-  const [attainmentTree, setAttainmentTree]: State<AttainmentData | null> =
-    useState(null);
-
-  const [instances, setInstances]: State<Array<CourseInstanceData>> = useState([]);
+  const [attainmentTree, setAttainmentTree]: State<AttainmentData | null | undefined> =
+    useState(undefined);
```

Comment:
Why both `null` and `undefined`?

Reply:
My approach was that its undefined when we are fetching the data from API, then we display loading. If we get no assessment models, set it to null to display "no attainments" message, otherwise render attainments.

Reply:
OK, you should add a comment explaining that.